db/user.go

```diff
@@ -221,3 +222,74 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// ResolveUsers is a helper function to map a User.Role to all User.Email's for that course.
+// The function takes an optional course and a list of strings, containing emails, roles, and * as input.
+// The function returns an alphanumerically sorted slice of lowercase emails without duplicates.
+// The function returns an error if a course cannot be opened and the input contains roles or a *.
+// The function will issue a warning if it's given a valid course and an invalid role.
```

Comment:
The function comment need a little work.
 - The first line is a little factually off (since a string could be a *, and it is never actually a role (just a string representing a role).
 - Since we do not actually have users here, don't use User.* (e.g. User.Role and User.Email).
 - I don't like the "The function ..." four times in a row. Try it a little more naturally. "This function takes ... and returns ... . On an error ... . A warning will be logged if ... ."